Formula/p/projectm.rb

```diff
@@ -20,59 +21,100 @@ class Projectm < Formula
+  end
+
   def install
-    system "./configure", "--disable-silent-rules", *std_configure_args
-    system "make", "install"
+    resource("projectM-eval").stage do
+      system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+      system "cmake", "--build", "build"
+      system "cmake", "--install", "build"
+    end
+
+    system "cmake", "-S", ".", "-B", "build",
+                               "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                               "-DBUILD_SHARED_LIBS=ON",
+                               *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    # remove openGL requirement because that doesn't work on macOS or our linux install
+    inreplace lib/"pkgconfig/projectM-4.pc", "Requires: opengl", ""
+
+    # fix library naming
+    inreplace lib/"pkgconfig/projectM-4.pc", "-l:projectM", "-lprojectM"
+    inreplace lib/"pkgconfig/projectM-4-playlist.pc", "-l:projectM", "-lprojectM"
   end
 
   test do
-    assert_path_exists prefix/"share/projectM/config.inp"
-    assert_path_exists prefix/"share/projectM/presets"
-
-    (testpath/"test.cpp").write <<~CPP
-      #include <libprojectM/projectM.hpp>
+    (testpath/"test.c").write <<~C
       #include <SDL2/SDL.h>
+
+      #include <projectM-4/projectM.h>
+
       #include <stdlib.h>
       #include <stdio.h>
 
-      int main()
+      int main(void)
       {
-        // initialize SDL video + openGL
-        if (SDL_Init(SDL_INIT_VIDEO) < 0)
-        {
-          fprintf(stderr, "Video init failed: %s", SDL_GetError());
-          return 1;
-        }
-        atexit(SDL_Quit);
-
-        SDL_Window *win = SDL_CreateWindow("projectM Test", 0, 0, 320, 240,
-                                          SDL_WINDOW_OPENGL | SDL_WINDOW_ALLOW_HIGHDPI);
-        SDL_GLContext glCtx = SDL_GL_CreateContext(win);
-
-        auto *settings = new projectM::Settings();
-        auto *pm = new projectM(*settings, projectM::FLAG_DISABLE_PLAYLIST_LOAD);
-
-        // if we get this far without crashing we're in good shape
-        return 0;
+          if (SDL_Init(SDL_INIT_VIDEO) < 0)
+          {
+              fprintf(stderr, "Video init failed: %s", SDL_GetError());
+              return 1;
+          }
+
+          SDL_Window* win = SDL_CreateWindow("projectM Test", 0, 0, 320, 240,
+              SDL_WINDOW_OPENGL | SDL_WINDOW_ALLOW_HIGHDPI);
+          if (win == NULL)
+          {
+              fprintf(stderr, "SDL Window creation failed: %s", SDL_GetError());
+              return 2;
+          }
+
+          SDL_GLContext glCtx = SDL_GL_CreateContext(win);
+          if (glCtx == NULL)
+          {
+              fprintf(stderr, "SDL GL context creation failed: %s", SDL_GetError());
+              return 3;
+          }
+
+          projectm_handle pm = projectm_create();
+          if (pm == NULL)
+          {
+              fprintf(stderr, "projectM instance creation failed: %s", SDL_GetError());
```

Comment:
@kblaschke  is this the correct function to get information on the project-m failure here?